apps/api/src/domain/models/song/commands/create-song.command.integration.spec.ts

```diff
@@ -69,64 +93,47 @@ describe('CreateSong', () => {
 
     describe('when the payload is valid', () => {
         it('should succeed', async () => {
-            const newId = await idManager.generate();
+            await assertCommandSuccess(assertionHelperDependencies, {
+                buildValidCommandFSA,
+                initialState,
+                checkStateOnSuccess: async ({ id }: CreateSong) => {
+                    const idStatus = await idManager.status(id);
 
-            const result = await commandHandlerService.execute(buildValidCommandFSA(newId));
-
-            expect(result).toBe(Ack);
-        });
+                    expect(idStatus).toBe(NotAvailable);
 
-        it('should persist the new instance', async () => {
-            const newId = await idManager.generate();
+                    const songSearchResult = await testRepositoryProvider
+                        .forResource<Song>(ResourceType.song)
+                        .fetchById(id);
 
-            await commandHandlerService.execute(buildValidCommandFSA(newId));
+                    expect(songSearchResult).not.toBe(NotFound);
 
-            const songSearchResult = await testRepositoryProvider
-                .forResource<Song>(ResourceType.song)
-                .fetchById(newId);
+                    expect(songSearchResult).not.toBeInstanceOf(InternalError);
 
-            expect(songSearchResult).not.toBe(NotFound);
-
-            expect(songSearchResult).not.toBeInstanceOf(InternalError);
-
-            expect((songSearchResult as Song).id).toBe(newId);
-        });
-
-        it('should mark the id as used', async () => {
-            const newId = await idManager.generate();
-
-            await commandHandlerService.execute(buildValidCommandFSA(newId));
-
-            const idStatus = await idManager.status(newId);
-
-            expect(idStatus).toBe(NotAvailable);
+                    expect((songSearchResult as Song).id).toBe(id);
+                },
+            });
         });
     });
 
     describe('when the payload has an invalid type', () => {
         describe('when the id property has an invalid type (number[])', () => {
             it('should return an error', async () => {
-                const newId = await idManager.generate();
-
-                const validCommandFSA = buildValidCommandFSA(newId);
-
-                const result = await commandHandlerService.execute({
-                    ...validCommandFSA,
-                    payload: {
-                        ...validCommandFSA.payload,
-                        id: [99],
+                await assertCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: (id: AggregateId) => buildInvalidFSA(id, { id: [99] }),
```

Comment:
The way this is called is a bit of a smell. Let's work on fixing the type for `buildCommandFSA`.